autoload/elixir/util.vim

```diff
@@ -26,23 +26,25 @@ function! elixir#util#count_indentable_symbol_diff(line, open, close)
   if elixir#util#is_indentable_match(a:line, a:open)
         \ && elixir#util#is_indentable_match(a:line, a:close)
     return
-          \   s:match_count(a:line.text, a:open)
-          \ - s:match_count(a:line.text, a:close)
+          \   s:match_count(a:line, a:open)
+          \ - s:match_count(a:line, a:close)
   else
     return 0
   end
 endfunction
 
-function! s:match_count(string, pattern)
-  let size = strlen(a:string)
+function! s:match_count(line, pattern)
+  let size = strlen(a:line.text)
   let index = 0
   let counter = 0
 
   while index < size
-    let index = match(a:string, a:pattern, index)
+    let index = match(a:line.text, a:pattern, index)
     if index >= 0
       let index += 1
-      let counter +=1
+      if elixir#util#is_indentable_at(a:line.num, index)
```

Comment:
Actually, the `is_indentable_at` uses the `column` and not the index. So your code is right in increment the index before checking. However, this means that the `is_indentable_match` function has a bug :(